### ISSUE TITLE
ldap: missing default values make configuration difficult

### ISSUE THREAD

**Issue by ThomasADavis** [bootc/netbox-chart#59]
You have to set a password for the ldap access..  in anonymous access, there is no need for a read-only connection to have a password.  This is the mode we run our ldap service as.

The chart errors out at:

```
[root@mouse-r13 netbox]# helm upgrade -n netbox netbox bootc/netbox -f ldap.yaml 
Error: UPGRADE FAILED: template: netbox/templates/secret.yaml:25:64: executing "netbox/templates/secret.yaml" at <b64enc>: invalid value; expected string
```

**Comment by ThomasADavis** [comment]
I did get it to work with anonymous access, and that's not documented either in the docker page..

**Comment by bootc** [comment]
What did you need to do to enable anonymous access? Set the field to the empty string?

**Comment by bootc** [comment]
I should add this is mostly a duplicate of #41 aside from the anonymous access question.

**Comment by ThomasADavis** [comment]
This is what worked; but yes, both binddn and bindpw have to be defined, and empty - aka ''

```
remoteAuth:
  enabled: true
    #backend: netbox.authentication.RemoteUserBackend
  backend: netbox.authentication.LDAPBackend
  autoCreateUser: true
    # header: HTTP_REMOTE_USER
  defaultGroups: []
  defaultPermissions: {}
  ldap:
    serverUri: 'ldaps://ldapcrt.nersc.gov'
    startTls: false
    ignoreCertErrors: true
    bindDn: ''
    bindPassword: ''
    userDnTemplate: null
    userSearchBaseDn: 'ou=people,ou=nim-ldap,ou=Host,o=ldapsvc,dc=nersc,dc=gov'
    userSearchAttr: 'uid'
    groupSearchBaseDn: 'ou=PosixGroup,ou=nim-ldap,ou=Host,o=ldapsvc,dc=nersc,dc=gov'
    groupSearchClass: 'posixGroup'
    groupType: 'PosixGroupType'
    requireGroupDn: 'cn=staff,ou=PosixGroup,ou=nim-ldap,ou=Host,o=ldapsvc,dc=nersc,dc=gov'
    findGroupPerms: true
    mirrorGroups: true
    mirrorGroupsExcept: null
    cacheTimeout: 3600
    isAdminDn: 'cn=staff,ou=PosixGroup,ou=nim-ldap,ou=Host,o=ldapsvc,dc=nersc,dc=gov'
    isSuperUserDn: 'cn=otg,ou=PosixGroup,ou=nim-ldap,ou=Host,o=ldapsvc,dc=nersc,dc=gov'
    attrFirstName: 'givenName'
    attrLastName: 'sn'
    attrMail: 'mail'
```

**Comment by bootc** [comment]
Thanks, I'll get a documentation tweak into the README that says that.

**Comment by bootc** [comment]
I've just released version [4.0.1](https://github.com/bootc/netbox-chart/releases/tag/4.0.1) of this chart which resolves this issue.
